volatility3/framework/__init__.py

```diff
@@ -69,6 +69,22 @@ def require_interface_version(*args) -> None:
 class Deprecation:
     """Deprecation related methods."""
 
+    @staticmethod
+    def method_being_removed(message: str):
+
+        def decorator(deprecated_func):
+            @functools.wraps(deprecated_func)
+            def wrapper(*args, **kwargs):
+                warnings.warn(
+                    f"This API ({deprecated_func.__module__}.{deprecated_func.__qualname__}) will be removed in a release very soon. {message}",
```

Comment:
Might be worth adding an extra parameter with a version number to say when it'll definitely be deprecated by?  Problem is the versions bump at unexpected times, so perhaps a datetime for removal is better?  It would also make it easy for us to grep them all at release and remove the ones whose time has come?